File: solutions/java/src/ridesharingservice/Ride.java
package ridesharingservice;

public class Ride {
    private final int id;
    private final Passenger passenger;
    private Driver driver;
    private final Location source;
    private final Location destination;
    private RideStatus status;
    private double fare;

    public Ride(int id, Passenger passenger, Driver driver, Location source, Location destination, RideStatus status, double fare) {
        this.id = id;
        this.passenger = passenger;
        this.driver = driver;
        this.source = source;
        this.destination = destination;
        this.status = status;
        this.fare = fare;
    }

    public void setDriver(Driver driver) {
        this.driver = driver;
    }

    public void setStatus(RideStatus status) {
        this.status = status;
    }

    public void setFare(double fare) {
        this.fare = fare;
    }

    public int getId() {
        return id;
    }

    public Passenger getPassenger() {
        return passenger;
    }

    public Driver getDriver() {
        return driver;
    }

    public Location getSource() {
        return source;
    }

    public Location getDestination() {
        return destination;
    }

    public RideStatus getStatus() {
        return status;
    }

    public double getFare() {
        return fare;
    }
}

File: solutions/csharp/ridesharingservice/Ride.cs
namespace RideSharingService
{
    public class Ride
    {
        public int Id { get; }
        public Passenger Passenger { get; }
        public Driver Driver { get; private set; }
        public Location Source { get; }
        public Location Destination { get; }
        public RideStatus Status { get; private set; }
        public double Fare { get; private set; }

        public Ride(int id, Passenger passenger, Driver driver, Location source, Location destination, RideStatus status, double fare)
        {
            Id = id;
            Passenger = passenger;
            Driver = driver;
            Source = source;
            Destination = destination;
            Status = status;
            Fare = fare;
        }

        public void SetDriver(Driver driver)
        {
            Driver = driver;
        }

        public void SetStatus(RideStatus status)
        {
            Status = status;
        }

        public void SetFare(double fare)
        {
            Fare = fare;
        }
    }
}